### PR TITLE
JAMES-2481 Upgrade apache.httpcomponents (httpclient-osgi & fluent-hc)

### PR DESCRIPTION
**Pull request by trantienduchn**

### CODE REVIEW

**Comment by trantienduchn** [comment]
rebase master

**Comment by trantienduchn** [comment]
🍏

**Comment by chibenwa** [comment]
```
[768931e942a4005dc9a4eb3b079b84162b056c27] Tests in error: 
[768931e942a4005dc9a4eb3b079b84162b056c27]   SpamAssassinContract.imapCopiesToSpamMailboxShouldBeConsideredAsSpam:183 » ConditionTimeout
```

test this please <3

File: pom.xml [hunk 1]
```diff
@@ -1878,12 +1878,12 @@
...
                 <groupId>javax.servlet</groupId>
                 <artifactId>javax.servlet-api</artifactId>
-                <version>3.1.0</version>
+                <version>4.0.1</version>
             </dependency>
             <dependency>
                 <groupId>javax.servlet</groupId>
                 <artifactId>servlet-api</artifactId>
-                <version>${servlet-api.version}</version>
+                <version>3.0-alpha-1</version>
```
Comment (mbaechler) [inline pom.xml]: which artifact depends on that ?

**Comment by chibenwa** [comment]
```
[768931e942a4005dc9a4eb3b079b84162b056c27] Tests in error: 
[768931e942a4005dc9a4eb3b079b84162b056c27]   VacationIntegrationTest.jmapVacationShouldNotSendNotificationTwice:237 » ConditionTimeout
```

test this please

File: pom.xml (hunk 1 above)
Comment (chibenwa) [reply pom.xml]: Nothing aparently....

**Comment by aduprat** [comment]
`spark` version & `servlet` version seems to be incompatible:
```
java.lang.NoSuchMethodError: javax.servlet.http.HttpServletResponse.getContentType()Ljava/lang/String;
	at spark.http.matching.Body.serializeTo(Body.java:64)
	at spark.http.matching.MatcherFilter.doFilter(MatcherFilter.java:189)
	at spark.embeddedserver.jetty.JettyHandler.doHandle(JettyHandler.java:50)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1568)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.eclipse.jetty.server.Server.handle(Server.java:531)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:352)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:281)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:102)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:762)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:680)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by aduprat** [comment]
test this please (not automatically scheduled by Jenkins on update)

File: server/container/filesystem-api/src/test/java/org/apache/james/filesystem/api/AbstractFileSystemTest.java [hunk 2]
```diff
@@ -54,24 +62,47 @@
...
     
     protected abstract FileSystem buildFileSystem(String configurationRootDirectory);
 
     @Before
     public void setUp() throws Exception {
-        httpServer = Service.ignite().port(RANDOM_PORT);
-        httpServer.get("/", (Request request, Response response) -> "content");
-        httpServer.awaitInitialization();
-        
+        httpServer = new Server(RANDOM_PORT);
```
Comment (mbaechler) [inline server/container/filesystem-api/src/test/java/org/apache/james/filesystem/api/AbstractFileSystemTest.java]: you should have a look at JettyHttpServerTest, you'll find that you don't need that much boilerplate (:

**Comment by aduprat** [comment]
```
[341] Running shell script
+ docker exec james-server-jpa-james-2481-1eada48e-d504-45a2-b70e-9d793a9f4a2e java -jar /root/james-cli.jar -h 127.0.0.1 adddomain domain
Error response from daemon: Container dcc2ce3872fdbe611eeba2af9c4284380d17df1b6fbede59c54898ca7c741710 is not running
```
Fail in JPA integration tests...

**Comment by chibenwa** [comment]
Same as explained yesterday: it's due to a mismatch in javaagent version. Can be safely considered :green_apple:

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**